### DISCUSSION TITLE
What is namespace for When() method while applying conditional branching in elsa 3.0

### DISCUSSION THREAD

**Discussion by NiveditaGalwade** [elsa-workflows/elsa-core#5595] [Q&A]
What is namespace for When() method while applying conditional branching in elsa 3.0 
my CustomWorkflow is like:

public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .Then<Activity2>()
            // Add a conditional activity here
            .Add<If>(activity => activity
                .WithCondition(context => context.GetInput<bool>("Flag"))
                .When(OutcomeNames.True).Then<Activity3>() // This will execute if the flag is true
                .When(OutcomeNames.False).Then<Activity4>()) // This will execute if the flag is false
            .Then<Activity5>(); // Continue with Activity5 after the conditional branch
    }
}

getting below error
Error CS1061 'ISetupActivity' does not contain a definition for 'When' and no accessible extension method 'When' accepting a first argument of type 'ISetupActivity' could be found (are you missing a using directive or an assembly reference?)

**Comment by glime-ai** [comment]
To use the `When` method for conditional branching in Elsa 3.0, you need to ensure that you are using the correct namespaces and that you have the appropriate version of Elsa that supports the method you are trying to use. In Elsa 3.0, the `When` method is typically used with the `Switch` or `If` activities to branch the workflow based on conditions.

From the context you've provided, it seems like you are trying to use the `When` method with an `If` activity. However, the `If` activity does not have a `When` method. Instead, you should use the `Then` method to specify the activities to execute for the true and false branches of the condition.

Here's how you can modify your workflow to use the `If` activity correctly:

```csharp
public class CustomWorkflow : IWorkflow
{
    public void Build(IWorkflowBuilder builder)
    {
        builder
            .StartWith<Activity1>()
            .Then<Activity2>()
            // Add a conditional activity here
            .If(context => context.GetInput<bool>("Flag"), @if =>
            {
                @if
                    .Then<Activity3>() // This will execute if the flag is true
                    .Else<Activity4>(); // This will execute if the flag is false
            })
            .Then<Activity5>(); // Continue with Activity5 after the conditional branch
    }
}
```

In this corrected example, the `If` activity is used with a lambda expression that defines the condition. The `Then` method is used to specify the activity to execute if the condition is true, and the `Else` method is used to specify the activity to execute if the condition is false.

Make sure you have the following namespaces included in your file:

```csharp
using Elsa.Builders;
using Elsa.Services.Models;
```

If you are still encountering issues, please ensure that you have the correct version of Elsa installed and that you are following the latest documentation for Elsa 3.0.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
